### PR TITLE
Merges the Export Scanner, Sales Tagger, and Price Tagger into one item.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

The Export Scanner, Sales Tagger, and Price Tagger are all niche, cargo adjacent items that each perform a different minor task and if using one, requires usually one of the three extremely rarely. Each one requires printing a new item at the lathe, all for maybe one or two uses before it needs to be replaced.

This PR merges the three into a singular, modular item. Existing export scanners have been reflavored the "universal scanner". When used in-hand, it produces a radial to swap between the three modes, Each mode has identical function to their previous scanner, with the only functional difference being that the sales tagger now prints new tags on secondary attack as opposed to primary.

The other two items lathe recipes have been removed, as their functionality has been fully moved to the universal scanner.

Also flips the export scanner icon state to match the other 2 scanner items that have been lumped together.

## Why It's Good For The Game

Cargo now only needs to swap between 1 of three nearly identical barcode scanner looking items at a time, as opposed to having to print an item that sees functional use once every thousand rounds. By increasing the ease of use, it will also make it easier to use custom vendors, as well as print sales tags on regular rounds, as the universal scanner can still be found in shipping boxes located around the station and in tool storages.

Arguably not a balance change, as the three tools are extremely cheap to print, and don't have a formal impact on cargo balance except for tedium.

## Changelog

:cl:
qol: The Export Scanner, Sales Tagger, and Price Tagger have been merged together into the Universal Scanner.
qol: Using the Universal Scanner in-hand allows for switching between the three tools functionality on the fly.
/:cl:

### CODE REVIEW

**Comment by tgsAvylaar** [comment]
Very nice!

File: code/modules/cargo/export_scanner.dm [hunk 1]
```diff
@@ -1,22 +1,159 @@
...
+	/// Minimum value for cut_multiplier in sales tagger mode.
+	var/cut_min = 0.01
+
+/obj/item/export_scanner/Initialize(mapload)
+	. = ..()
+	scale_mode = sort_list(list(
+		"export scanner" = image(icon = src.icon, icon_state = "export scanner"),
+		"price tagger" = image(icon = src.icon, icon_state = "price tagger"),
+		"sales tagger" = image(icon = src.icon, icon_state = "sales tagger")
+		))
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:39]:
```suggestion
		"sales tagger" = image(icon = src.icon, icon_state = "sales tagger"),
	))
```

File: code/modules/cargo/export_scanner.dm [hunk 2]
```diff
@@ -1,22 +1,159 @@
+
 /obj/item/export_scanner
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:2]: could you change the path to reflect the object better for admins or mappers or whatever?

File: code/modules/cargo/export_scanner.dm [hunk 3]
```diff
@@ -1,22 +1,159 @@
+
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:1]: ?

File: code/modules/cargo/export_scanner.dm [hunk 4]
```diff
@@ -1,22 +1,159 @@
...
+		if("price tagger")
+			scanning_mode = SCAN_PRICE_TAG
+		if("sales tagger")
+			scanning_mode = SCAN_SALES_TAG
+	icon_state = "[choice]"
+	playsound(src, 'sound/machines/click.ogg', 40, TRUE)
+
+
 
 /obj/item/export_scanner/afterattack(obj/O, mob/user, proximity)
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:60]: if youre touching the proc make it use better variable names please

File: code/modules/cargo/export_scanner.dm [hunk 5]
```diff
@@ -1,22 +1,159 @@
...
+/obj/item/export_scanner/attack_self_secondary(mob/user, modifiers)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
+		if(paper_count <= 0)
+			to_chat(user, span_warning("You're out of paper!'."))
+			return
+		if(!payments_acc)
+			to_chat(user, span_warning("You need to swipe [src] with an ID card first."))
+			return
+		paper_count -= 1
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:106]:
```suggestion
		paper_count--
```

File: code/modules/cargo/export_scanner.dm [hunk 6]
```diff
@@ -1,22 +1,159 @@
...
+			else
+				payments_acc = potential_acc.registered_account
+				playsound(src, 'sound/machines/ping.ogg', 40, TRUE)
+				to_chat(user, span_notice("[src] registers the ID card. Tag a wrapped item to create a barcode."))
+		else if(!potential_acc.registered_account)
+			to_chat(user, span_warning("This ID card has no account registered!"))
+			return
+	if(istype(attacking_item, /obj/item/paper))
+		if (!(paper_count >= max_paper_count))
+			paper_count += 10
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:87]: define or variable this

File: code/modules/cargo/export_scanner.dm [hunk 7]
```diff
@@ -1,22 +1,159 @@
...
+		user.put_in_hands(new_barcode)
+	if(scanning_mode == SCAN_PRICE_TAG)
+		if(loc != user)
+			to_chat(user, span_warning("You must be holding \the [src] to continue!"))
+			return
+		var/chosen_price = tgui_input_number(user, "Set price", "Price", new_custom_price)
+		if(!chosen_price || QDELETED(user) || QDELETED(src) || !user.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE) || loc != user)
+			return
+		new_custom_price = chosen_price
+		to_chat(user, span_notice(" The [src] will now give things a [new_custom_price] cr tag."))
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:121]:
```suggestion
		to_chat(user, span_notice("The [src] will now give things a [new_custom_price] cr tag."))
```

File: code/modules/cargo/export_scanner.dm [hunk 8]
```diff
@@ -1,22 +1,159 @@
...
+
+/obj/item/export_scanner/CtrlClick(mob/user)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
+		payments_acc = null
+		to_chat(user, span_notice("You clear the registered account."))
+
+/obj/item/export_scanner/AltClick(mob/user)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:131]: early return

File: code/modules/cargo/export_scanner.dm [hunk 9]
```diff
@@ -1,22 +1,159 @@
...
+		var/potential_cut = input("How much would you like to pay out to the registered card?","Percentage Profit ([round(cut_min*100)]% - [round(cut_max*100)]%)") as num|null
+		if(!potential_cut)
+			cut_multiplier = initial(cut_multiplier)
+		cut_multiplier = clamp(round(potential_cut/100, cut_min), cut_min, cut_max)
+		to_chat(user, span_notice("[round(cut_multiplier*100)]% profit will be received if a package with a barcode is sold."))
+
+/obj/item/export_scanner/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] has [paper_count]/[max_paper_count] available barcodes. Refill with paper.")
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:140]: i think "It" will be better than the object name, youre already examining the object

File: code/modules/cargo/export_scanner.dm [hunk 10]
```diff
@@ -1,22 +1,159 @@
...
+		if(payments_acc)
+			. += span_notice("<b>Ctrl-click</b> to clear the registered account.")
+
+	if(scanning_mode == SCAN_PRICE_TAG)
+		. += span_notice("The current custom price is set to [new_custom_price] cr.")
+
+/**
+ * Scans an object, O, and providesit's export value based on selling to the cargo shuttle, to mob/user.
+ */
+/obj/item/export_scanner/proc/export_scan(obj/O, mob/user)
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:153]: same here make this cleaner please

File: code/modules/cargo/export_scanner.dm [hunk 11]
```diff
@@ -1,22 +1,159 @@
...
+	if(scanning_mode == SCAN_SALES_TAG)
+		. += span_notice("Profit split on sale is currently set to [round(cut_multiplier*100)]%. <b>Alt-click</b> to change.")
+		if(payments_acc)
+			. += span_notice("<b>Ctrl-click</b> to clear the registered account.")
+
+	if(scanning_mode == SCAN_PRICE_TAG)
+		. += span_notice("The current custom price is set to [new_custom_price] cr.")
+
+/**
+ * Scans an object, O, and providesit's export value based on selling to the cargo shuttle, to mob/user.
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:151]: dont forget about the comment

File: code/modules/cargo/export_scanner.dm [hunk 12]
```diff
@@ -48,3 +185,35 @@
 
 			else
 				to_chat(user, span_warning("Bank account not detected. Handling tip not registered."))
+
+/**
+ * Scans an object, target, and sets it's custom_price variable to new_custom_price, presenting it to the user.
+ */
+/obj/item/export_scanner/proc/price_tag(obj/target, mob/user)
+	if(isitem(target))
+		var/obj/item/I = target
```
Comment (Fikou) [inline code/modules/cargo/export_scanner.dm:194]: better variable name

**Review by Fikou — CHANGES_REQUESTED** [review]
also you need to add tooltips for the different things you can press (i dont think you have stuff on right click, you could move some alt click functionality there too

File: code/modules/cargo/export_scanner.dm [hunk 13]
```diff
@@ -1,34 +1,173 @@
...
+			scanning_mode = SCAN_EXPORTS
+		if("price tagger")
+			scanning_mode = SCAN_PRICE_TAG
+		if("sales tagger")
+			scanning_mode = SCAN_SALES_TAG
+	icon_state = "[choice]"
+	playsound(src, 'sound/machines/click.ogg', 40, TRUE)
+
+
+
```
Comment (dragomagol) [inline code/modules/cargo/export_scanner.dm:60]:
```suggestion

```

File: code/modules/cargo/export_scanner.dm [hunk 14]
```diff
@@ -1,34 +1,173 @@
-/obj/item/export_scanner
-	name = "export scanner"
-	desc = "A device used to check objects against Nanotrasen exports database."
+#define PAPER_PER_SHEET 10
+
+/obj/item/universal_scanner
```
Comment (dragomagol) [inline code/modules/cargo/export_scanner.dm:3]: Did you still want this file to be called export_scanner?

**Comment by DAKKA-WAAAGH** [comment]
Where picture?

**Comment by ArcaneMusic** [comment]
> Where picture?

The sprite change is extremely small, namely renaming a few sprites and flipping another single sprite.
image

**Comment by moocowswag** [comment]
This is not a qol pr, this is a balance change because it limits the accessibility of custom vendors which are reliant on price taggers to function.

**Pull request merged**